ddl/ddl.go

```diff
@@ -797,6 +799,27 @@ func (d *ddl) DoDDLJob(ctx sessionctx.Context, job *model.Job) error {
 			return context.Canceled
 		}
 
+		// If the connection being killed, we need to CANCEL the DDL job.
+		if atomic.LoadUint32(&sessVars.Killed) == 1 {
+			if sessVars.StmtCtx.DDLJobID != 0 {
```

Comment:
If we do killing in DDL, then can we just use `jobID` instead of `StmtCtx`?

Reply:
This operation is intend to avoid repeating the `CancelDDL` calling.
`sessVars.StmtCtx.DDLJobID = 0` means maybe we have executed `CancelDDL` already.

Reply:
Got